File: impares.m
clc; clear all; close all;
pkg load database;
retry= true;

while retry
  try
    fprintf("\n Elija una opcion \n 1.Correr el programa \n 2.Mostrar Historial \n");
    m=(input('Elija:   '));
    if m>2||m==0|m<0
      fprintf("Elija una opcion correcta \n");
    endif
    if m==1
      cont=1;
      for x=1:2:100
          V(cont)=x;
          cont=cont+1;
      endfor
      
      fprintf("La cantidad de numero imapares es: %d \n",cont-1);
      disp(V);

      s=cont-1;
      save('impares.txt','V','s','-ascii');
      params= cell(1,2);
      ss=mat2str(V);
      params{1,1}=ss;
      params{1,2}=s;
      conn = pq_connect(setdbopts('dbname','HWP','host','localhost',
      'port','5432','user','postgres','password','123fgthg'));
      N=pq_exec_params(conn, "insert into impar1(imapres,cantidad) values($1,$2);",params); %insertar datos en la tabla
      save('impares.txt','V','-ascii');
      retry=false;1
    endif
    if m==2
      conn = pq_connect(setdbopts('dbname','HWP','host','localhost',
      'port','5432','user','postgres','password','123fgthg'));
      N=pq_exec_params(conn, "Select * from impar1;"); %insertar datos en la tabla
      disp(N)
      retry=false;
    endif
    
  catch
    fprintf("A ingresado un valor erroneo vuelva a intentarlo %d \n");
    msg = lasterror.message;
    fprintf(msg);
  end_try_catch


endwhile

File: impares.py
import psycopg2
from tabulate import tabulate
def  post(x,y):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "HWP"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO impar1(imapres, cantidad) VALUES (%s, %s);", (x, y))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "HWP"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from impar1;")
    print(tabulate(cursor, headers=["ID", "IMPARES", "CANTIDAD"], tablefmt="fancy_grid", numalign ="center"))

print("Numeros impares del 1 al 100 \n")
r=[]
while True:
    try:
        print("Elija una opcion:\n1.Correr el Programa\n2.Mostrar el Historial\n3.Salir")
        b=int(input("Elija:\n"))
        if b>3 or b<=0:
            print("Valor ingresado fuera del rango del menu.\n")
        elif b==1:
            for x in range(1,100,2):
                r.append(x)
            print("Los numero impares del 1 al 100 son: \n ",r)
            post(r,len(r))
            file = open("C:/Users/Roberto/Documents/Proyectos/TAREA 1/impares.txt","w")
            file.write("Los numero impares del 1 al 100 son: \n "+ str(r))
            file.close()
        elif b==2:
            ret()
        elif b==3:
            break

    except Exception as e:
        print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
        print(repr(e))

exit(0)